cavern/src/intTest/java/org/opencadc/cavern/FilesPutTest.java

```diff
@@ -0,0 +1,318 @@
+            Assert.fail("Unexcepted exception: " + t.getMessage());
+        }
+    }
+
+
+    @Test
+    public void testPutOverwriteFileOK() throws Throwable {
+        final String uri = putTestURIStr + "/smallTextFile2.rtf";
+        final VOSURI nodeURI = new VOSURI(uri);
+
+        // Make a DataNode of the same name as the test file,
+        // PUT that node using /cavern/nodes service
+        Subject.doAs(cadcauthSubject, new PrivilegedExceptionAction() {
+            @Override
+            public Object run() throws Exception {
+                DataNode dn = new DataNode(nodeURI);
+                dn.getProperties().add(new NodeProperty(VOS.PROPERTY_URI_ISPUBLIC, Boolean.FALSE.toString()));
+                putNode(new URL(nodesURL.toExternalForm() + dn.getUri().getPath()), dn);
+                return null;
+            }
+        });
+
+        // Now attempt to PUT the file using /cavern/files - should still work
+        try {
+            Subject.doAs(cadcauthSubject, new PrivilegedAction<Object>() {
+                public Object run() {
+                    try {
+                        String filename = "src/test/resources/smallTextFile2.rtf";
+                        putFile(nodeURI, filename);
+                        return null;
+                    } catch (Throwable t) {
+                        throw new RuntimeException(t);
+                    }
+                }
+            });
+
+        } catch (Throwable t) {
+            log.error(t.getMessage(), t);
+            Assert.fail("Unexcepted exception: " + t.getMessage());
+        }
+    }
+
+
+    @Test
+    public void testPutFileNotAuth() throws Throwable {
+        final String uri = putTestURIStr + "/smallTextFile.rtf";
+        final VOSURI nodeURI = new VOSURI(uri);
+
+        try {
+            String filename = "src/test/resources/smallTextFile.rtf";
+            putFile(nodeURI, filename);
+            Assert.fail("should have had an access error.");
+
+        } catch (AccessControlException t) {
+            log.debug("expected & got AccessControlException");
+
+        } catch (Throwable t) {
+            log.error(t.getMessage(), t);
+            Assert.fail("Unexcepted exception: " + t.getMessage());
+        }
+    }
+
+
+    private void putFile(VOSURI uri, String fileName) throws Throwable {
+
+        RegistryClient regClient = new RegistryClient();
+        AuthMethod authMethod = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
+        if (authMethod == null) {
+            authMethod = AuthMethod.ANON;
+        }
+        URL filePutURL = regClient.getServiceURL(uri.getServiceURI(), Standards.VOSPACE_FILES_20, authMethod);
+        log.debug("baseURL for getFile: " + filePutURL.toExternalForm());
+        URL url = new URL(filePutURL.toString() + uri.getPath());
+        log.debug("requested url for putFile: " + url.toExternalForm());
+
+        File fileToUpload = new File(fileName);
```

Comment:
You could create this file in java.io.tmp to be consistent.  Not sure where this goes or what the 'current working directory' is.  Or not, probably file like this.